File: user-service/src/test/java/rs/raf/user_service/unit/ActuaryServiceTest.java
package rs.raf.user_service.unit;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.*;
import org.springframework.data.domain.*;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.scheduling.annotation.Scheduled;
import rs.raf.user_service.client.BankClient;
import rs.raf.user_service.client.StockClient;
import rs.raf.user_service.domain.dto.ActuaryDto;
import rs.raf.user_service.domain.dto.ActuaryLimitDto;
import rs.raf.user_service.domain.dto.EmployeeDto;
import rs.raf.user_service.domain.entity.ActuaryLimit;
import rs.raf.user_service.domain.entity.Employee;
import rs.raf.user_service.domain.entity.Role;
import rs.raf.user_service.domain.mapper.EmployeeMapper;
import rs.raf.user_service.exceptions.ActuaryLimitNotFoundException;
import rs.raf.user_service.exceptions.EmployeeNotFoundException;
import rs.raf.user_service.exceptions.UserNotAgentException;
import rs.raf.user_service.repository.ActuaryLimitRepository;
import rs.raf.user_service.repository.EmployeeRepository;
import rs.raf.user_service.service.ActuaryService;

import java.math.BigDecimal;
import java.util.*;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.ArgumentMatchers.*;
import static org.mockito.Mockito.*;

@ExtendWith(org.mockito.junit.jupiter.MockitoExtension.class)
public class ActuaryServiceTest {

    @Mock
    private ActuaryLimitRepository actuaryLimitRepository;

    @Mock
    private EmployeeRepository employeeRepository;

    @Mock
    private StockClient stockClient;

    @InjectMocks
    private ActuaryService actuaryService;

    private Employee agentEmployee;
    private Employee nonAgentEmployee;
    private Role agentRole;
    private Role adminRole;
    private ActuaryLimit actuaryLimit;

    @BeforeEach
    void setUp() {
        // Inicijalizacija test podataka
        agentRole = new Role();
        agentRole.setName("AGENT");

        adminRole = new Role();
        adminRole.setName("ADMIN");

        agentEmployee = new Employee();
        agentEmployee.setId(1L);
        agentEmployee.setRole(agentRole);

        nonAgentEmployee = new Employee();
        nonAgentEmployee.setId(2L);
        nonAgentEmployee.setRole(adminRole);

        actuaryLimit = new ActuaryLimit();
        actuaryLimit.setEmployee(agentEmployee);  // Umesto setEmployeeId(...)
        actuaryLimit.setLimitAmount(BigDecimal.valueOf(1000));
        actuaryLimit.setUsedLimit(BigDecimal.ZERO);
        actuaryLimit.setNeedsApproval(false);
    }

    // findAll(...) ------------------------------------------------------------------------------------------------

    @Test
    void testFindAll_Success() {
        // Podesimo mock da vrati neku listu zaposlenih (agentEmployee)
        Page<Employee> employeesPage = new PageImpl<>(List.of(agentEmployee));
        Pageable pageable = PageRequest.of(0, 10);
        when(employeeRepository.findAll(any(Specification.class), eq(pageable)))
                .thenReturn(employeesPage);

        // Poziv servisa
        Page<EmployeeDto> result = actuaryService.findAgents("John", "Doe", "agent@example.com", "AGENT", pageable);

        // Provera rezultata
        assertNotNull(result);
        assertEquals(1, result.getTotalElements());
        // Proveravamo da li je metoda za pronalazak pozvana
        verify(employeeRepository, times(1)).findAll(any(Specification.class), eq(pageable));
    }

    // changeAgentLimit(...) ---------------------------------------------------------------------------------------

    @Test
    void testChangeAgentLimit_Success() {
        Long employeeId = 1L;
        BigDecimal newLimit = BigDecimal.valueOf(2000);

        // Podesimo mockove: nađemo zaposlenog i ActuaryLimit
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.of(actuaryLimit));

        // Izvršimo metodu
        actuaryService.changeAgentLimit(employeeId, newLimit);

        // Proverimo da li je ActuaryLimit sačuvan sa novim limitom
        assertEquals(newLimit, actuaryLimit.getLimitAmount());
        verify(actuaryLimitRepository, times(1)).save(actuaryLimit);
    }

    @Test
    void testChangeAgentLimit_EmployeeNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.empty());

        assertThrows(EmployeeNotFoundException.class, () ->
                actuaryService.changeAgentLimit(employeeId, BigDecimal.valueOf(1000))
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testChangeAgentLimit_UserNotAgent() {
        Long employeeId = 2L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(nonAgentEmployee));

        assertThrows(UserNotAgentException.class, () ->
                actuaryService.changeAgentLimit(employeeId, BigDecimal.valueOf(500))
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testChangeAgentLimit_ActuaryLimitNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.empty());

        assertThrows(ActuaryLimitNotFoundException.class, () ->
                actuaryService.changeAgentLimit(employeeId, BigDecimal.valueOf(300))
        );
        verify(actuaryLimitRepository, never()).save(any(ActuaryLimit.class));
    }

    // resetDailyLimit(...) ---------------------------------------------------------------------------------------

    @Test
    void testResetDailyLimit_Success() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.of(actuaryLimit));

        actuaryService.resetDailyLimit(employeeId);

        assertEquals(BigDecimal.ZERO, actuaryLimit.getUsedLimit());
        verify(actuaryLimitRepository, times(1)).save(actuaryLimit);
    }

    @Test
    void testResetDailyLimit_EmployeeNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.empty());

        assertThrows(EmployeeNotFoundException.class, () ->
                actuaryService.resetDailyLimit(employeeId)
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testResetDailyLimit_UserNotAgent() {
        Long employeeId = 2L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(nonAgentEmployee));

        assertThrows(UserNotAgentException.class, () ->
                actuaryService.resetDailyLimit(employeeId)
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testResetDailyLimit_ActuaryLimitNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.empty());

        assertThrows(ActuaryLimitNotFoundException.class, () ->
                actuaryService.resetDailyLimit(employeeId)
        );
        verify(actuaryLimitRepository, never()).save(any(ActuaryLimit.class));
    }

    // setApproval(...) -------------------------------------------------------------------------------------------

    @Test
    void testSetApproval_Success() {
        Long employeeId = 1L;
        boolean approvalValue = true;

        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.of(actuaryLimit));

        actuaryService.setApproval(employeeId, approvalValue);

        assertTrue(actuaryLimit.isNeedsApproval());
        verify(actuaryLimitRepository, times(1)).save(actuaryLimit);
    }

    @Test
    void testSetApproval_EmployeeNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.empty());

        assertThrows(EmployeeNotFoundException.class, () ->
                actuaryService.setApproval(employeeId, true)
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testSetApproval_UserNotAgent() {
        Long employeeId = 2L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(nonAgentEmployee));

        assertThrows(UserNotAgentException.class, () ->
                actuaryService.setApproval(employeeId, false)
        );
        verify(actuaryLimitRepository, never()).findByEmployeeId(anyLong());
    }

    @Test
    void testSetApproval_ActuaryLimitNotFound() {
        Long employeeId = 1L;
        when(employeeRepository.findById(employeeId)).thenReturn(Optional.of(agentEmployee));
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.empty());

        assertThrows(ActuaryLimitNotFoundException.class, () ->
                actuaryService.setApproval(employeeId, true)
        );
        verify(actuaryLimitRepository, never()).save(any(ActuaryLimit.class));
    }

    // getAgentLimit(...) -----------------------------------------------------------------------------------------

    @Test
    void testGetAgentLimit_Success() {
        Long employeeId = 1L;
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.of(actuaryLimit));

        ActuaryLimitDto dto = actuaryService.getAgentLimit(employeeId);

        assertNotNull(dto);
        assertEquals(actuaryLimit.getLimitAmount(), dto.getLimitAmount());
        assertEquals(actuaryLimit.getUsedLimit(), dto.getUsedLimit());
        assertEquals(actuaryLimit.isNeedsApproval(), dto.isNeedsApproval());
        verify(actuaryLimitRepository, times(1)).findByEmployeeId(employeeId);
    }

    @Test
    void testGetAgentLimit_NotFound() {
        Long employeeId = 999L;
        when(actuaryLimitRepository.findByEmployeeId(employeeId)).thenReturn(Optional.empty());

        assertThrows(ActuaryLimitNotFoundException.class, () ->
                actuaryService.getAgentLimit(employeeId)
        );
    }

    // resetDailyLimits() -----------------------------------------------------------------------------------------

    @Test
    void testResetDailyLimits_Success() {
        // Ovde testiramo "scheduled" metodu koja svakodnevno resetuje limit
        ActuaryLimit limit1 = new ActuaryLimit();
        limit1.setUsedLimit(BigDecimal.valueOf(100));

        ActuaryLimit limit2 = new ActuaryLimit();
        limit2.setUsedLimit(BigDecimal.valueOf(200));

        List<ActuaryLimit> limits = List.of(limit1, limit2);
        when(actuaryLimitRepository.findAll()).thenReturn(limits);

        actuaryService.resetDailyLimits();

        assertEquals(BigDecimal.ZERO, limit1.getUsedLimit());
        assertEquals(BigDecimal.ZERO, limit2.getUsedLimit());
        // Proveravamo da li je sve snimljeno
        verify(actuaryLimitRepository, times(1)).saveAll(limits);
    }

}

File: user-service-kotlin/src/test/kotlin/pack/userservicekotlin/service/ActuaryServiceTest.kt
package pack.userservicekotlin.service

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.*
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.domain.Specification
import pack.userservicekotlin.arrow.ActuaryServiceError
import pack.userservicekotlin.domain.TestDataFactory
import pack.userservicekotlin.domain.entities.Client
import pack.userservicekotlin.domain.entities.Employee
import pack.userservicekotlin.external.StockClient
import pack.userservicekotlin.repository.ActuaryLimitRepository
import pack.userservicekotlin.repository.ClientRepository
import pack.userservicekotlin.repository.EmployeeRepository
import java.math.BigDecimal
import java.util.*

@ExtendWith(MockitoExtension::class)
class ActuaryServiceTest {
    @Mock
    lateinit var employeeRepository: EmployeeRepository

    @Mock
    lateinit var actuaryLimitRepository: ActuaryLimitRepository

    @Mock
    lateinit var clientRepository: ClientRepository

    @Mock
    lateinit var stockClient: StockClient

    @Mock
    lateinit var pageable: Pageable

    @InjectMocks
    lateinit var actuaryService: ActuaryService

    @Test
    fun `findAll returns page of EmployeeResponseDto`() {
        val employee = TestDataFactory.employee()
        val employeePage = PageImpl(listOf(employee))

        `when`(
            employeeRepository.findAll(
                argThat<Specification<Employee>> {
                    it is Specification<Employee>
                },
                eq(pageable),
            ),
        ).thenReturn(employeePage)

        val result = actuaryService.findAll("a", "b", "c", "d", pageable)

        assertEquals(1, result.totalElements)
        assertEquals(employee.firstName, result.content[0].firstName)
    }

    @Test
    fun `changeAgentLimit succeeds`() {
        val employee = TestDataFactory.employee(id = 1L)
        val limit = TestDataFactory.actuaryLimit()
        employee.actuaryLimit = limit

        `when`(employeeRepository.findById(1L)).thenReturn(Optional.of(employee))
        `when`(actuaryLimitRepository.findByEmployeeId(1L)).thenReturn(Optional.of(limit))

        val result = actuaryService.changeAgentLimit(1L, BigDecimal(9000))

        assertTrue(result.isRight())
        assertEquals(BigDecimal(9000), limit.limitAmount)
        verify(actuaryLimitRepository).save(limit)
    }

    @Test
    fun `changeAgentLimit fails when employee not found`() {
        `when`(employeeRepository.findById(1L)).thenReturn(Optional.empty())

        val result = actuaryService.changeAgentLimit(1L, BigDecimal(9000))

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.EmployeeNotFound)
    }

    @Test
    fun `resetDailyLimit sets usedLimit to zero`() {
        val employee = TestDataFactory.employee(id = 2L)
        val limit = TestDataFactory.actuaryLimit(usedLimit = BigDecimal(100))
        employee.actuaryLimit = limit

        `when`(employeeRepository.findById(2L)).thenReturn(Optional.of(employee))
        `when`(actuaryLimitRepository.findByEmployeeId(2L)).thenReturn(Optional.of(limit))

        val result = actuaryService.resetDailyLimit(2L)

        assertTrue(result.isRight())
        assertEquals(BigDecimal.ZERO, limit.usedLimit)
        verify(actuaryLimitRepository).save(limit)
    }

    @Test
    fun `resetDailyLimit fails when employee not found`() {
        `when`(employeeRepository.findById(2L)).thenReturn(Optional.empty())

        val result = actuaryService.resetDailyLimit(2L)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.EmployeeNotFound)
    }

    @Test
    fun `setApproval sets needsApproval`() {
        val employee = TestDataFactory.employee(id = 3L)
        val limit = TestDataFactory.actuaryLimit(needsApproval = false)
        employee.actuaryLimit = limit

        `when`(employeeRepository.findById(3L)).thenReturn(Optional.of(employee))
        `when`(actuaryLimitRepository.findByEmployeeId(3L)).thenReturn(Optional.of(limit))

        val result = actuaryService.setApproval(3L, true)

        assertTrue(result.isRight())
        assertTrue(limit.needsApproval)
        verify(actuaryLimitRepository).save(limit)
    }

    @Test
    fun `getAgentLimit returns response DTO`() {
        val limit = TestDataFactory.actuaryLimit()

        `when`(actuaryLimitRepository.findByEmployeeId(4L)).thenReturn(Optional.of(limit))

        val result = actuaryService.getAgentLimit(4L)

        assertTrue(result.isRight())
        val dto = result.getOrNull()
        assertEquals(limit.limitAmount, dto?.limitAmount)
        assertEquals(limit.usedLimit, dto?.usedLimit)
        assertEquals(limit.needsApproval, dto?.needsApproval)
    }

    @Test
    fun `getAgentLimit fails when limit not found`() {
        `when`(actuaryLimitRepository.findByEmployeeId(4L)).thenReturn(Optional.empty())

        val result = actuaryService.getAgentLimit(4L)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.ActuaryLimitNotFound)
    }

    @Test
    fun `resetDailyLimits resets all used limits`() {
        val limit1 = TestDataFactory.actuaryLimit(usedLimit = BigDecimal(999))
        val limit2 = TestDataFactory.actuaryLimit(usedLimit = BigDecimal(888))

        `when`(actuaryLimitRepository.findAll()).thenReturn(listOf(limit1, limit2))

        actuaryService.resetDailyLimits()

        assertEquals(BigDecimal.ZERO, limit1.usedLimit)
        assertEquals(BigDecimal.ZERO, limit2.usedLimit)
        verify(actuaryLimitRepository).saveAll(listOf(limit1, limit2))
    }

    @Test
    fun `findActuaries returns enriched page`() {
        val employee = TestDataFactory.employee(id = 1L)
        val employeePage = PageImpl(listOf(employee))
        val order = TestDataFactory.orderDto(userId = 1L, profit = BigDecimal(100))

        `when`(
            employeeRepository.findAll(
                argThat<Specification<Employee>> {
                    it is Specification<Employee>
                },
                eq(pageable),
            ),
        ).thenReturn(employeePage)
        `when`(stockClient.getAll()).thenReturn(listOf(order))

        val result = actuaryService.findActuaries(pageable)

        assertTrue(result.isRight())
        val page = result.getOrNull()
        assertEquals(1, page?.totalElements)
        assertEquals(BigDecimal(100), page?.content?.get(0)?.profit)
    }

    @Test
    fun `findActuaries fails when stock client throws exception`() {
        val employee = TestDataFactory.employee(id = 1L)
        val employeePage = PageImpl(listOf(employee))

        `when`(
            employeeRepository.findAll(
                argThat<Specification<Employee>> {
                    it is Specification<Employee>
                },
                eq(pageable),
            ),
        ).thenReturn(employeePage)
        `when`(stockClient.getAll()).thenThrow(RuntimeException("Stock service down"))

        val result = actuaryService.findActuaries(pageable)

        assertTrue(result.isLeft())
        val error = result.swap().getOrNull()
        assertTrue(error is ActuaryServiceError.ExternalServiceError)
    }

    @Test
    fun `findAgents returns page of AgentDto`() {
        val employee = TestDataFactory.employee(id = 1L)
        val employeePage = PageImpl(listOf(employee))
        val limit = TestDataFactory.actuaryLimit()

        `when`(
            employeeRepository.findAll(
                argThat<Specification<Employee>> {
                    it is Specification<Employee>
                },
                eq(pageable),
            ),
        ).thenReturn(employeePage)
        `when`(actuaryLimitRepository.findByEmployeeId(1L)).thenReturn(Optional.of(limit))

        val result = actuaryService.findAgents("a", "b", "c", "d", pageable)

        assertTrue(result.isRight())
        val page = result.getOrNull()
        assertEquals(1, page?.totalElements)
        assertEquals(limit.limitAmount, page?.content?.get(0)?.limitAmount)
        assertEquals(limit.usedLimit, page?.content?.get(0)?.usedLimit)
        assertEquals(limit.needsApproval, page?.content?.get(0)?.needsApproval)
    }

    @Test
    fun `findAgents fails when actuary limit not found`() {
        val employee = TestDataFactory.employee(id = 1L)
        val employeePage = PageImpl(listOf(employee))

        `when`(
            employeeRepository.findAll(
                argThat<Specification<Employee>> {
                    it is Specification<Employee>
                },
                eq(pageable),
            ),
        ).thenReturn(
            @Suppress("ktlint:standard:max-line-length")
            employeePage,
        )
        `when`(actuaryLimitRepository.findByEmployeeId(1L)).thenReturn(Optional.empty())

        val result = actuaryService.findAgents("a", "b", "c", "d", pageable)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.ActuaryLimitNotFound)
    }

    @Test
    fun `getAllAgentsAndClients returns combined list with role filter`() {
        val employee = TestDataFactory.employee(id = 1L, role = TestDataFactory.role("AGENT"))
        val client = TestDataFactory.client(id = 2L)

        `when`(employeeRepository.findAll(argThat<Specification<Employee>> { it is Specification<Employee> })).thenReturn(listOf(employee))
        `when`(clientRepository.findAll(argThat<Specification<Client>> { it is Specification<Client> })).thenReturn(emptyList())

        val result = actuaryService.getAllAgentsAndClients("", "", "AGENT")

        assertTrue(result.isRight())
        val list = result.getOrNull()
        assertEquals(1, list?.size)
        assertEquals("AGENT", list?.get(0)?.role)
    }

    @Test
    fun `getAllAgentsAndClients returns combined list with name filter`() {
        val employee = TestDataFactory.employee(id = 1L, firstName = "John")
        val client = TestDataFactory.client(id = 2L, firstName = "John")

        `when`(employeeRepository.findAll(argThat<Specification<Employee>> { it is Specification<Employee> })).thenReturn(listOf(employee))
        `when`(clientRepository.findAll(argThat<Specification<Client>> { it is Specification<Client> })).thenReturn(listOf(client))

        val result = actuaryService.getAllAgentsAndClients("John", "", "")

        assertTrue(result.isRight())
        val list = result.getOrNull()
        assertEquals(2, list?.size)
        assertTrue(list?.all { it.firstName == "John" } ?: false)
    }

    @Test
    fun `updateUsedLimit succeeds`() {
        val limit = TestDataFactory.actuaryLimit()
        val newLimit = BigDecimal(5000)

        `when`(actuaryLimitRepository.findByEmployeeId(1L)).thenReturn(Optional.of(limit))

        val result = actuaryService.updateUsedLimit(1L, newLimit)

        assertTrue(result.isRight())
        val dto = result.getOrNull()
        assertEquals(newLimit, dto?.usedLimit)
        assertEquals(limit.limitAmount, dto?.limitAmount)
        assertEquals(limit.needsApproval, dto?.needsApproval)
        verify(actuaryLimitRepository).save(limit)
    }

    @Test
    fun `updateUsedLimit fails when limit not found`() {
        `when`(actuaryLimitRepository.findByEmployeeId(1L)).thenReturn(Optional.empty())

        val result = actuaryService.updateUsedLimit(1L, BigDecimal(5000))

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.ActuaryLimitNotFound)
    }

    @Test
    fun `changeAgentLimit fails when employee is not an agent`() {
        val employee = TestDataFactory.employee(id = 1L, role = TestDataFactory.role("ADMIN"))

        `when`(employeeRepository.findById(1L)).thenReturn(Optional.of(employee))

        val result = actuaryService.changeAgentLimit(1L, BigDecimal(9000))

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.NotAnAgent)
    }

    @Test
    fun `resetDailyLimit fails when employee is not an agent`() {
        val employee = TestDataFactory.employee(id = 1L, role = TestDataFactory.role("ADMIN"))

        `when`(employeeRepository.findById(1L)).thenReturn(Optional.of(employee))

        val result = actuaryService.resetDailyLimit(1L)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.NotAnAgent)
    }

    @Test
    fun `setApproval fails when employee is not an agent`() {
        val employee = TestDataFactory.employee(id = 1L, role = TestDataFactory.role("ADMIN"))

        `when`(employeeRepository.findById(1L)).thenReturn(Optional.of(employee))

        val result = actuaryService.setApproval(1L, true)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.NotAnAgent)
    }

    @Test
    fun `findActuaries fails when pageable is null`() {
        val result = actuaryService.findActuaries(null)

        assertTrue(result.isLeft())
        assertTrue(result.swap().getOrNull() is ActuaryServiceError.NotAnAgent)
    }
}
